pkgs/test_core/lib/src/runner/configuration/args.dart

```diff
@@ -181,7 +181,15 @@ void _parseTestSelection(
   TestSelection selection;
   String path;
   if (firstQuestion == -1) {
-    path = option;
+    if (option.contains('://')) {
+      path = Uri.parse(option).path;
+      // Strip out the leading slash before the drive letter.
+      if (Platform.isWindows && path.startsWith('/')) {
+        path = path.substring(1);
+      }
+    } else {
+      path = option;
+    }
```

Comment:
Does it work to let it fall through to the other Uri handling path?

```dart
if (firstQuestion == -1 && !option.contains('://')) {
  path = option;
```

Reply:
I guess it might cause a problem with the first `else if`?

Maybe we could do

```dart
if (firstQuestion == -1 && !option.contains('://')) {
  path = option;
  selection = TestSelection();
} else {
  final uri = Uri.tryParse(option);
  if (uri == null) {
    throw FormatException(...);
  }
  // ...
}
```

Reply:
I unified the URI parsing, not using `tryParse` though, just letting it do its normal exception throwing for now.

Reply:
Ok I further simplified this now by always parsing it as a URI but then decoding the path component afterwords which restores any back slashes. This means we can support query params with backslash style paths also (even though its weird/invalid, it doesn't hurt?)

Reply:
Ok I don't understand why but I am now seeing encoded backslashes, at least for the new test I added.

I expanded that test to work on any platform as well as adding a new test which uses a query param with platform specific separators. If it is going to work, we might as well test it.